host/sgx/windows/enter.asm

```diff
@@ -103,11 +111,22 @@ execute_eenter:
 
     mov ARG1OUT, rdi
     mov ARG2OUT, rsi
-	
-    ;; Restore the saved MXCSR
-    ldmxcsr MXCSR
-	
+
+    ;; Restore the saved host context
+    mov rcx, HOST_CONTEXT
+    call oe_restore_host_context
+
 dispatch_ocall:
+
+    ;; Save callee-preserved registers
+    push rbx
```

Comment:
Do we need this? Technically the callee (`__oe_dispatch_ocall`) should preserve these registers right?

Reply:
No, you're right. It turns out the pushes were just masking a stack alignment bug prior to the `__oe_dispatch_ocall`.

I've clarified the alignment of the stack at the beginning of the function and corrected the offset for this pseudo-prologue for the call into `__oe_dispatch_ocall` to maintain that alignment.